central/role/datastore/defaults.go

```diff
@@ -0,0 +1,220 @@
+					permissions.Modify(resources.VulnerabilityReports), // required for vuln report configurations prior to collections
+				}
+			}
+			return []permissions.ResourceWithAccess{
+				permissions.View(resources.WorkflowAdministration),   // required for vuln report configurations
+				permissions.Modify(resources.WorkflowAdministration), // required for vuln report configurations
+				permissions.View(resources.Integration),              // required for vuln report configurations
+			}
+		}(),
+	},
+	accesscontrol.VulnerabilityManagementConsumer: {
+		idSuffix:    "vulnmgmtconsumer",
+		postgresID:  accesscontrol.DefaultPermissionSetIDs[accesscontrol.VulnerabilityManagementConsumer],
+		description: "For users: use it to provide read-only access to analyze vulnerabilities and initiate risk acceptance process",
+		resourceWithAccess: []permissions.ResourceWithAccess{
+			permissions.View(resources.Node),
+			permissions.View(resources.Deployment),
+			permissions.View(resources.Image),
+			permissions.View(resources.WatchedImage),
+			permissions.View(resources.VulnerabilityReports),
+			permissions.View(resources.WorkflowAdministration),
+			permissions.Modify(resources.VulnerabilityManagementRequests),
+		},
+	},
+	accesscontrol.VulnerabilityManagementAdmin: {
+		idSuffix:    "vulnmgmtadmin",
+		postgresID:  accesscontrol.DefaultPermissionSetIDs[accesscontrol.VulnerabilityManagementAdmin],
+		description: "For users: Use it to provide administrative access to analyze vulnerabilities, generate reports, and manage risk acceptance process",
+		resourceWithAccess: []permissions.ResourceWithAccess{
+			permissions.View(resources.Cluster),
+			permissions.View(resources.Node),
+			permissions.View(resources.Namespace),
+			permissions.View(resources.Deployment),
+			permissions.View(resources.Image),
+			permissions.View(resources.Integration),
+			permissions.Modify(resources.WatchedImage),
+			permissions.Modify(resources.VulnerabilityManagementRequests),
+			permissions.Modify(resources.VulnerabilityManagementApprovals),
+			permissions.Modify(resources.VulnerabilityReports),
+			permissions.Modify(resources.WorkflowAdministration),
+		},
+	},
+}
+
+func getDefaultRoles() []*storage.Role {
+	roles := make([]*storage.Role, 0, len(defaultPermissionSets))
+	for _, roleName := range accesscontrol.DefaultRoleNames.AsSlice() {
+		attributes, found := defaultPermissionSets[roleName]
+		if !found {
+			utils.Should(errors.Errorf("Default role %s does not have permission set defined", roleName))
+		}
+
+		// Historically, default permission sets and roles have had same name to make it easier for the user to quickly map one to the other.
+		permissionSet := getDefaultPermissionSet(roleName)
+		role := &storage.Role{
+			Name:          roleName,
+			Description:   attributes.description,
+			AccessScopeId: rolePkg.AccessScopeIncludeAll.GetId(),
+			Traits: &storage.Traits{
+				Origin: storage.Traits_DEFAULT,
+			},
+			PermissionSetId: permissionSet.Id,
```

Comment:
If `getDefaultPermissionSet` yields nil due to there being no name <-> ID match, then this would yield a nil pointer panic.

Either we handle the case `permissionSet == nil` explicitly, or we use `permissionSet.GetId()` here.

I'd be more in favor of using `utils.Should` in case of nil permission set and skip adding the role.